numba/tests/test_unicode.py

```diff
@@ -337,6 +350,54 @@ def test_find(self, flags=no_pyobj_flags):
                                  cfunc(a, substr),
                                  "'%s'.find('%s')?" % (a, substr))
 
+    def test_count(self):
+        pyfunc = count_usecase
+        cfunc = njit(pyfunc)
+
+        for s in UNICODE_EXAMPLES + ['aaaa']:
+            extras = ['', ' ', 'xx', s[::-1], s[:-2], s[3:], s, s + s, 'aa']
+            for sub in [x for x in extras]:
+                self.assertEqual(pyfunc(s, sub),
+                                 cfunc(s, sub),
+                                 "'%s' in '%s'?" % (sub, s))
+
+    def test_count_with_start_end(self):
+        pyfunc = count_with_start_end_usecase
+        cfunc = njit(pyfunc)
+
+        cpython = ['aaaa', '\u0102', '\U00100304', '\U00100304' + 'a_' * 10]
+        for s in UNICODE_EXAMPLES + cpython:
+            extras = ['', ' ', 'xx', s[::-1], s[:-2], s[3:], s, s + s, 'a', 'a_',
+                      '\u0102', 'a\u0102', 'a\U00100304', '\u0102_', '\u0102\U00100304']
+            for sub in [x for x in extras]:
+                for i , j in product(range(-10,10), (0,20)):
```

Comment:
```suggestion
                for i , j in product(range(-10, 10), (0, 20)):
```

Reply:
Also, might it make sense for these ranges extend symmetrically about zero and to the length of the longest test string+1 ? Same comment applies below to the `start only` case.

For example, right now, this fails and was not caught by this test:
```python
from numba import njit

@njit
def foo(x, y):
    return x.count(y, -40, -40)

a = "ascii"
b = ""

print(foo.py_func(a, b))
print(foo(a, b))
```
the implementation in this PR erroneously returns `6`.

Reply:
@stuartarchibald I am not clear about the range suggestion. So are you suggesting to have the range like (-20,20) or (0,20)?